utils/legacy_pointer/tests/offset.cc

```diff
@@ -0,0 +1,175 @@
+
+  void operator()() {
+    auto float_off = offset_ / sizeof(float);
+    float * ptr = reinterpret_cast<float *>(&*accB_.get_pointer());
+    ptr[float_off] = i_ * SIZE_ + j_;
+  };
+};
+
+TEST(offset, basic_test) {
+  {
+    ASSERT_EQ(legacy::getPointerMapper().count(), 0u);
+    float * myPtr = static_cast<float *>(
+                        legacy::malloc(100 * sizeof(float)));
+
+    ASSERT_NE(myPtr, nullptr);
+
+    ASSERT_FALSE(legacy::PointerMapper::is_nullptr(myPtr));
+
+    ASSERT_EQ(legacy::getPointerMapper().count(), 1u);
+
+    myPtr += 3;
+
+    // Obtain the buffer id
+    buffer_id bId = legacy::getPointerMapper().get_buffer_id(myPtr);
+    ASSERT_NE(bId, 0); // Only one buffer
+
+    // Obtain the buffer
+    // Note that the scope of this buffer ends when the buffer
+    // is freed
+    //
+    buffer_t b = legacy::getPointerMapper().get_buffer(bId);
+
+    size_t offset = legacy::getPointerMapper().get_offset(myPtr);
+    ASSERT_EQ(offset, 3*sizeof(float));
+    
+    cl::sycl::queue q;
+    q.submit([&b,offset](cl::sycl::handler& h) {
+        auto accB = b.get_access<sycl_acc_rw>(h);
+        h.single_task<class foo1>([=]() {
+              accB[offset] = 1.0f;
+            });
+        });
+
+    // Only way of reading the value is using a host accessor
+    {
+      auto hostAcc = b.get_access<sycl_acc_rw, sycl_acc_host>();
+      ASSERT_EQ(hostAcc[offset], 1.0f);
+    }
+    legacy::free(myPtr);
+    ASSERT_EQ(legacy::getPointerMapper().count(), 0u);
+  }
+}
+
+
+TEST(offset, 2d_indexing) {
+  {
+    const unsigned SIZE = 8;
+    ASSERT_EQ(legacy::getPointerMapper().count(), 0u);
+    float * myPtr = static_cast<float *>(
+                        legacy::malloc(SIZE * SIZE * sizeof(float)));
+
+    ASSERT_NE(myPtr, nullptr);
+
+    ASSERT_FALSE(legacy::PointerMapper::is_nullptr(myPtr));
+
+    ASSERT_EQ(legacy::getPointerMapper().count(), 1u);
+
+    // Obtain the buffer id
+    buffer_id bId = legacy::getPointerMapper().get_buffer_id(myPtr);
+    ASSERT_NE(bId, 0); // Only one buffer
+    
+    cl::sycl::queue q;
+
+    float * actPos = myPtr;
+    for (unsigned i = 0; i < SIZE; i++) {   
+      for (unsigned j = 0; j < SIZE; j++) {
+        // Obtain the buffer id
+        buffer_id bId = 
+          legacy::getPointerMapper().get_buffer_id(actPos);
+        ASSERT_NE(bId, 0); // Only one buffer
+
+        // Obtain the buffer
+        // Note that the scope of this buffer ends when the buffer
+        // is freed
+        //
+        buffer_t b = legacy::getPointerMapper().get_buffer(bId);
+
+        size_t offset = legacy::getPointerMapper().get_offset(actPos);
+        ASSERT_EQ(offset, (i * SIZE + j) * sizeof(float));
+        
+        q.submit([b, offset, i, j, SIZE](cl::sycl::handler& h) mutable {
+            auto accB = b.get_access<sycl_acc_rw>(h);
+            h.single_task(kernel(accB, i, j, SIZE, offset));
+        });
+        q.wait_and_throw();
```

Comment:
The queue has no async handler so there will be no throw here - could change to a wait, or add a handler